aten/src/ATen/native/Linear.cpp

```diff
@@ -825,6 +826,14 @@ Tensor tensordot(const Tensor& input1, const Tensor& input2, IntArrayRef dims1,
 }
 
 Tensor &tensordot_out(const Tensor& input1, const Tensor& input2, IntArrayRef dims1, IntArrayRef dims2, Tensor& result) {
+  if(result.defined()) {
+    TORCH_CHECK(
+      !(result.requires_grad() && at::GradMode::is_enabled()),
```

Comment:
Let's also limit this to the case where the shapes don't match

Reply:
(You'd need to move this check below after the output is computed)

Reply:
Thanks for the comment, I have further narrowed down the exception here :)